plugins/ptp_operator/metrics/logparser.go

```diff
@@ -271,14 +276,17 @@ func extractPTP4lEventState(output string, isFollowerOnly bool) (portID int, rol
 		return
 	}
 
-	portIndex := fields[1]
-	role = types.UNKNOWN
+	// Use regex to extract port ID safely, supporting both:
+	// - "port 1:"
+	// - "port 1 (ens2f0):"
+	match := portMatch.FindStringSubmatch(output)
```

Comment:
I don't see why a regex is needed here. If we used the old replacer the port ID should still be in the same index in the resulting slice from the string.Fields as the changes are after it.

I double checked here: https://go.dev/play/p/kHimtzjSlEN

Reply:
you are right , I can revert that back